parsec/core/fs/remote_loader.py

```diff
@@ -417,20 +425,51 @@ async def load_manifest(
             FSDeviceNotFoundError
             FSInvalidTrustchainError
         """
+        return await self._load_manifest(
+            entry_id,
+            version=version,
+            timestamp=timestamp,
+            expected_backend_timestamp=expected_backend_timestamp,
+        )
+
+    async def _load_manifest(
+        self,
+        entry_id: EntryID,
+        version: Optional[int] = None,
+        timestamp: Optional[DateTime] = None,
+        expected_backend_timestamp: Optional[DateTime] = None,
+        workspace_entry: Optional[WorkspaceEntry] = None,
+    ) -> BaseRemoteManifest:
         if timestamp is not None and version is not None:
             raise FSError(
                 f"Supplied both version {version} and timestamp `{timestamp}` for manifest "
                 f"`{entry_id}`"
             )
         # Download the vlob
-        workspace_entry = self.get_workspace_entry()
+        current_workspace_entry = self.get_workspace_entry()
+        if workspace_entry is None:
+            workspace_entry = self.get_workspace_entry()
```

Comment:
Why two call to `get_workspace_entry` ?
Given there is no `await` between the two, both call should return the same entry, but I think it would be better to do
```python
workspace_entry = workspace_entry or current_workspace_entry
```

Reply:
5769bda